core/tests/webdriverio_utils/BlogPages.js

```diff
@@ -0,0 +1,308 @@
+  var noResultsFoundContainer = $('.e2e-test-no-results-shown');
+  var oppiaBlogHomePageCardContainer = $('.e2e-test-oppia-blog-home-page-card');
+  var oppiaAuthorProfilePageCardContainer = $(
+    '.e2e-test-oppia-author-profile-page-card');
+  var blogHomepageWelcomeHeading = $('.e2e-test-blog-welcome-heading');
+  var oppiaAvatarImage = $('.e2e-test-oppia-avatar-image');
+  var oppiaBlogPostPageContainer = $('.e2e-test-oppia-blog-post-page-card');
+  var blogPostTitleContainer = $('.e2e-test-blog-post-page-title-container');
+  var authorProfilePhoto = $('.e2e-test-author-profile-photo');
+  var authorNameContainer = $('.e2e-test-author-name');
+  var blogPostTagFilter = $('.e2e-test-tag-filter-component');
+  var blogPostSearchField = $('.e2e-test-search-field');
+  var blogPostsList = $('.e2e-test-blog-post-list');
+  var navigateToBlogHomePageButton = $('.e2e-test-back-button');
+  var paginationNextButton = $('.e2e-test-pagination-next-button');
+  var paginationPrevButton = $('.e2e-test-pagination-prev-button');
+  var paginationContainer = $('.e2e-test-pagination');
+  var searchInput = $('.e2e-test-search-input');
+  var blogPostTilesSelector = function() {
+    return $$('.e2e-test-blog-post-tile-item');
+  };
+  var blogPostRecommendationTilesSelector = function() {
+    return $$('.e2e-test-blog-post-tile-item');
+  };
+
+  this.submitSearchQuery = async function(searchQuery) {
+    await waitFor.pageToFullyLoad();
+    await action.clear('Search input', searchInput);
+    await action.setValue('Search input', searchInput, searchQuery);
+  };
+
+  this.get = async function() {
+    await browser.url(general.BLOG_PAGE_URL_SUFFIX);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.getBlogPostSearchPage = async function(searchQuery) {
+    await browser.url(
+      general.BLOG_PAGE_SEARCH_URL_PREFIX + '?q=' + searchQuery);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.expectNoResultsFoundShown = async function() {
+    await this.waitForVisibilityOfBlogHomePageContainer();
+    await waitFor.visibilityOf(
+      noResultsFoundContainer,
+      'No results found container taking too long to display');
+  };
+
+  this.expectBlogHomePageWelcomeHeadingToBeVisible = async function() {
+    await waitFor.visibilityOf(
+      blogHomepageWelcomeHeading,
+      'Blog Home Page Heading taking too long to display');
+  };
+
+  this.expectOppiaAvatarImageToBeVisible = async function() {
+    await waitFor.visibilityOf(
+      oppiaAvatarImage,
+      'Oppia Image Avatar image taking too long to display');
+  };
+
+  this.expectTagFilterComponentToBeVisible = async function() {
+    await waitFor.visibilityOf(
+      blogPostTagFilter,
+      'Tag Filter component taking too long to display');
+  };
+
+  this.getBlogDashboardPage = async function() {
+    blogDashboardPage = new BlogDashboardPage.BlogDashboardPage();
+    await blogDashboardPage.get();
+    await blogDashboardPage.expectCurrUserToHaveProfilePhoto();
```

Comment:
The name `getBlogDashboardPage` doesn't sound like it performs any assertions, so the `expectCurrUserToHaveProfilePhoto` function should be called separately

Reply:
Removed the function as it was not required.